sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveClientSuite.scala

```diff
@@ -122,6 +122,22 @@ class HiveClientSuite(version: String)
       "aa" :: Nil)
   }
 
+  test("getPartitionsByFilter: cast(chunk as int)=1 (not a valid partition predicate)") {
```

Comment:
I ran this test with current master branch, seems it can also pass without this change?

Reply:
yes this can pass, we can tell it by #12586

`Cast.mayTruncate` returns true for string to int, but not string to boolean. Here I just wanna test both of the cases, to improve test coverage.